gen-resourcesdocs/README.md

```diff
@@ -0,0 +1,124 @@
+  - affinity.podAffinity (PodAffinity) <-- link to definition   |    Affinity
+                                                                | composite type
+    Describes pod affinity scheduling rules (e.g. co-locate this|
+    pod in the same node, zone, etc. as some other pod(s)).     |
+                                                                |
+  - affinity.podAntiAffinity (PodAntiAffinity) <-- link to def. |
+                                                                |
+    Describes pod anti-affinity scheduling rules (e.g. avoid    |
+    putting this pod in the same node, zone, etc. as some other |
+    pod(s)).                                                  --/
+```
+
+In this example, the `affinity` field type is `Affinity`, which is a composite type. Because the `Affinity` definition is not listed in any of the `otherDefinitions`, this definition is documented inline.
+
+The `nodeAffinity` field type of the `Affinity` composite type is `NodeAffinity`, which is listed in a `otherDefinitions`. For this reason, this composite type is not documented inline, but the `NodeAffinity` is a link to the definition elsewhere in the documentation. Same for the `podAffinity` and `PodAntiAffinity` fields.
+
+### Ordering and categorization of fields
+
+By default, the fields of a Definition are listed in alphabetic order. The `content/<version>/fields.yaml` file is used to order the fields in another order, and also to group fields into categories.
+
+```yaml
+- definition: io.k8s.api.core.v1.PodSpec
+  field_categories:
+  - name: Containers
+    fields:
+    - containers
+    - initContainers
+    - imagePullSecrets
+    - enableServiceLinks
+  - name: Volumes
+    fields:
+    - volumes
+  - name: Scheduling
+    fields:
+    - nodeSelector
+    - nodeName
+    - affinity
+    - tolerations
+    - schedulerName
+    - runtimeClassName
+    - priorityClassName
+    - priority
+  [...]
+```
+
+In this example, the fields of the `PodSpec` definition are grouped and ordered in *Containers*, *Volumes*, *Scheduling* and other categories.
+
+Note that if a Definition appears in the `fields.yaml` file and some fields of the definition do not appear in the list of fields, the program will indicate that these fields are missing.
+
+The `name` attribute of `field_categories` in `fields.yaml` is optional. You can omit `name` when you want either to specify the order of the fields for a Definition  without creating a category, or to place some fields outside of any category, before other categories.
+
+## Common Definitions
+
+The `toc.yaml` file defines a **Common Definitions** part, containing a list of Definitions that are used in different places in the Kubernetes API.
+
+This way, a composite type can be documented in three places:
+
+- inline where this composite type is used (by default),
+- in a specific chapter, if the Definition is listed in `otherDefinitions`,
```

Comment:
I am not clear on "otherDefinitions"?

Reply:
Yes, it creates a new page on the Common definitions section for this definition, for example https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/exec-action/

It is rendered with a H1 heading actually